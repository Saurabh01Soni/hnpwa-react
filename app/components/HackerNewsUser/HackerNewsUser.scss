.HackerNewsUser {
  padding: 2em;
}

.HackerNewsUser__links {
  margin-top: 1em;

  a {
    color: $color-black;
  }
}
