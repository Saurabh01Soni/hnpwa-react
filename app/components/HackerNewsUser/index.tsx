import { Map } from 'immutable';
import * as React from 'react';

import LoadingIndicator from 'components/LoadingIndicator';

import './HackerNewsUser.scss';

interface IHackerNewsUserProps {
  readonly done?: () => void;
  readonly information?: Map<string, any> | null;
  readonly user: string;
  readonly onUserFetch?: (arg: string) => Promise<any>;
}

class HackerNewsUser extends React.Component<IHackerNewsUserProps> {
  public static defaultProps: IHackerNewsUserProps = {
    done: () => {},
    information: null,
    user: '',
    onUserFetch: (arg: string) => Promise.resolve(),
  };

  public componentWillMount() {
    const {
      done,
      information,
      user,
      onUserFetch,
    } = this.props;

    if (!information) {
      onUserFetch(user).then(done, done);
    }
  }

  public render() {
    const { information, user } = this.props;

    if (!information) {
      return (
        <LoadingIndicator
          active
          style={{ position: 'absolute', left: 'calc(50% - 24px)', marginTop: '30px' }}
        />
      );
    }

    return (
      <div className="HackerNewsUser">
        <table>
          <tbody>
            <tr>
              <td>User: </td>
              <td>{user}</td>
            </tr>
            <tr>
              <td>Created: </td>
              <td>{information.get('created')}</td>
            </tr>
            <tr>
              <td>Karma: </td>
              <td>{information.get('karma')}</td>
            </tr>
          </tbody>
        </table>
        <div className="HackerNewsUser__links">
          <a href={`https://news.ycombinator.com/submitted?id=${user}`}>submissions</a>
          {' | '}
          <a href={`https://news.ycombinator.com/threads?id=${user}`}>comments</a>
          {' | '}
          <a href={`https://news.ycombinator.com/favorites?id=${user}`}>favorites</a>
        </div>
      </div>
    );
  }
}

export default HackerNewsUser;
