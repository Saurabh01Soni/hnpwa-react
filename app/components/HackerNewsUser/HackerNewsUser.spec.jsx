import Immutable from 'immutable';
import React from 'react';
import { shallow } from 'enzyme';

import HackerNewsUser from './';

describe('<HackerNewsUser />', () => {
  it('should match snapshot when render default', () => {
    const wrapper = shallow(<HackerNewsUser user="taehwanno" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render user view when information exist', () => {
    const information = Immutable.Map({ created: '3 years ago', karma: 1234 });
    const wrapper = shallow(<HackerNewsUser information={information} user="taehwanno" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should calls props.onUserFetch when calls componentWillMount', () => {
    const onUserFetch = jest.fn();
    shallow(<HackerNewsUser user="taehwanno" onUserFetch={onUserFetch} />);
    expect(onUserFetch.mock.calls.length).toBe(1);
    expect(onUserFetch.mock.calls[0]).toEqual(['taehwanno']);
  });
});
