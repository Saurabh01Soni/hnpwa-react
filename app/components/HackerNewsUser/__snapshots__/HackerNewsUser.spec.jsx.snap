// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HackerNewsUser /> should match snapshot when render default 1`] = `""`;

exports[`<HackerNewsUser /> should render user view when information exist 1`] = `
<div
  className="HackerNewsUser"
>
  <table>
    <tbody>
      <tr>
        <td>
          User: 
        </td>
        <td>
          taehwanno
        </td>
      </tr>
      <tr>
        <td>
          Created: 
        </td>
        <td>
          3 years ago
        </td>
      </tr>
      <tr>
        <td>
          Karma: 
        </td>
        <td>
          1234
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="HackerNewsUser__links"
  >
    <a
      href="https://news.ycombinator.com/submitted?id=taehwanno"
    >
      submissions
    </a>
     | 
    <a
      href="https://news.ycombinator.com/threads?id=taehwanno"
    >
      comments
    </a>
     | 
    <a
      href="https://news.ycombinator.com/favorites?id=taehwanno"
    >
      favorites
    </a>
  </div>
</div>
`;
